persistent-test/src/PersistentTestModels.hs

```diff
@@ -133,8 +137,32 @@ deriving instance Eq (BackendKey backend) => Eq (NoPrefix1Generic backend)
 deriving instance Show (BackendKey backend) => Show (NoPrefix2Generic backend)
 deriving instance Eq (BackendKey backend) => Eq (NoPrefix2Generic backend)
 
-deriving instance Show (BackendKey backend) => Show (RelationshipGeneric backend)
-deriving instance Eq (BackendKey backend) => Eq (RelationshipGeneric backend)
+share [mkPersist persistSettings { 
+          mpsFieldLabelModifier = \entity field -> case entity of
+            "customPrefix1" -> append "_cp1" field
+            "customPrefix2" -> append "_cp2" field
```

Comment:
It is - IMO - surprising that the `HaskellName` for the *table* is lowercased! I would assume that that parameter would be identical to what's on the table. One thing I love about tests is that they surface this kind of thing.